From dac330136368a9b8d9ccf8227f56ea35de57a4d2 Mon Sep 17 00:00:00 2001
From: Paulo Alcantara <pc@manguebit.com>
Date: Fri, 8 Mar 2024 13:25:22 -0300
Subject: [PATCH] pam_cifscreds: fix warning on NULL arg passed to %s in
 pam_syslog()
MIME-Version: 1.0
Content-Type: text/plain; charset=UTF-8
Content-Transfer-Encoding: 8bit

Fix the following compiler warning with -Wformat-overflow in
cifscreds_pam_update():

  pam_cifscreds.c: In function ‘cifscreds_pam_update’:
  pam_cifscreds.c:340:83: warning: ‘%s’ directive argument is null
  [-Wformat-overflow=]
    340 | pam_syslog(ph, LOG_ERR, "error: Update credential key for %s: %s",
        |                                                           ^~

Fixes: cbbcd6e71c0a ("cifscreds: create PAM module to insert credentials at login")
Signed-off-by: Paulo Alcantara (Red Hat) <pc@manguebit.com>
---
 pam_cifscreds.c | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/pam_cifscreds.c b/pam_cifscreds.c
index 5d99c2db3038..eb9851d52a7d 100644
--- a/pam_cifscreds.c
+++ b/pam_cifscreds.c
@@ -338,7 +338,7 @@ static int cifscreds_pam_update(pam_handle_t *ph, const char *user, const char *
 		key_serial_t key = key_add(currentaddress, user, password, keytype);
 		if (key <= 0) {
 			pam_syslog(ph, LOG_ERR, "error: Update credential key for %s: %s",
-				currentaddress, strerror(errno));
+				   (currentaddress ?: "(null)"), strerror(errno));
 		}
 	}
 
-- 
2.44.0

